seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/client/DorisSinkManager.java

```diff
@@ -0,0 +1,161 @@
+        }
+        initialize = true;
+
+        if (batchIntervalMs != null) {
+            scheduler = Executors.newSingleThreadScheduledExecutor(
+                    new ThreadFactoryBuilder().setNameFormat("Doris-sink-output-%s").build());
+            scheduledFuture = scheduler.scheduleAtFixedRate(
+                () -> {
+                    try {
+                        flush();
+                    } catch (IOException e) {
+                        flushException = e;
+                    }
+                },
+                    batchIntervalMs,
+                    batchIntervalMs,
+                    TimeUnit.MILLISECONDS);
+        }
+    }
+
+    public synchronized void write(String record) throws IOException {
+        tryInit();
+        checkFlushException();
+        byte[] bts = record.getBytes(StandardCharsets.UTF_8);
+        batchList.add(bts);
+        batchRowCount++;
+        batchBytesSize += bts.length;
+        if (batchRowCount >= sinkConfig.getBatchMaxSize() || batchBytesSize >= sinkConfig.getBatchMaxBytes()) {
+            flush();
+        }
+    }
+
+    public synchronized void close() throws IOException {
+        if (scheduledFuture != null) {
+            scheduledFuture.cancel(false);
+            scheduler.shutdown();
+        }
+
+        flush();
+    }
+
+    public synchronized void flush() throws IOException {
+        checkFlushException();
+        if (batchList.isEmpty()) {
+            return;
+        }
+        String label = createBatchLabel();
+        DorisFlushTuple tuple = new DorisFlushTuple(label, batchBytesSize, new ArrayList<>(batchList));
+        for (int i = 0; i <= sinkConfig.getMaxRetries(); i++) {
+            try {
+                Boolean successFlag = dorisStreamLoadVisitor.doStreamLoad(tuple);
+                if (successFlag) {
+                    break;
+                }
+            } catch (Exception e) {
+                log.warn("Writing records to Doris failed, retry times = {}", i, e);
+                if (i >= sinkConfig.getMaxRetries()) {
+                    throw new DorisConnectorException(DorisConnectorErrorCode.WRITE_RECORDS_FAILED, "The number of retries was exceeded,writing records to Doris failed.", e);
+                }
+
+                if (e instanceof DorisConnectorException && ((DorisConnectorException) e).needReCreateLabel()) {
+                    String newLabel = createBatchLabel();
+                    log.warn(String.format("Batch label changed from [%s] to [%s]", tuple.getLabel(), newLabel));
+                    tuple.setLabel(newLabel);
+                }
+
+                try {
+                    long backoff = Math.min(sinkConfig.getRetryBackoffMultiplierMs() * i,
+                            sinkConfig.getMaxRetryBackoffMs());
+                    Thread.sleep(backoff);
+                } catch (InterruptedException ex) {
+                    Thread.currentThread().interrupt();
+                    throw new IOException(
```

Comment:
Unified exception

Reply:
Thanks for your advice. I have fixed it. Please review it again.